packages/react-reconciler/src/ReactFiberCompleteWork.js

```diff
@@ -521,7 +522,17 @@ function preloadInstanceAndSuspendIfNeeded(
   renderLanes: Lanes,
 ) {
   // Ask the renderer if this instance should suspend the commit.
-  if (!maySuspendCommit(type, props)) {
+  if (workInProgress.memoizedState !== null) {
```

Comment:
Should probably split this function in two and hoist the `maySuspend` stuff into `completeWork` so that this check is not in the shared path.

The suspendy logic at the bottom of this function can remain.

So something like:

```js
const maySuspend = checkInstanceMaySuspend(...);

// Then right before existing completeWork
if (maySuspend) {
  preloadInstanceAndSuspendIfNeeded(...);
}
```